gossipd/routing.c

```diff
@@ -1007,6 +1033,11 @@ bool routing_add_channel_announcement(struct routing_state *rstate,
 		delete_chan_messages_from_store(rstate, oldchan);
 		free_chans_from_node(rstate, oldchan);
 		tal_free(oldchan);
+		/* Effectively delete private update from db */
+		tell_lightningd_private_update(rstate, &node_id_1, scid,
+					       0, 0, 0, amount_msat(0), amount_msat(0), 0, 0);
+		tell_lightningd_private_update(rstate, &node_id_2, scid,
+					       0, 0, 0, amount_msat(0), amount_msat(0), 0, 0);
```

Comment:
I don't think this is necessary?  Once the channel is no longer private, this information is irrelevant, but in the meantime it's better than nothing.